packages/events/src/event.rs

```diff
@@ -176,4 +259,88 @@ mod test {
 
         assert!(!event_without_contract_address.is_from_contract(&contract_address));
     }
+
+    #[test]
+    fn block_begin_event_conversion_should_succeed() {
+        let height: u64 = 12345;
+        let proto_event =
+            ampd_proto::subscribe_response::Event::BlockBegin(ampd_proto::EventBlockBegin {
+                height,
+            });
+
+        let domain_event_response = Event::try_from(proto_event);
+        assert!(domain_event_response.is_ok());
+        let domain_event = domain_event_response.unwrap();
+
+        assert!(
+            matches!(domain_event, Event::BlockBegin(h) if h == block::Height::try_from(height).unwrap())
+        );
+    }
+
+    #[test]
+    fn block_end_event_conversion_should_succeed() {
+        let height: u64 = 54321;
+        let proto_event =
+            ampd_proto::subscribe_response::Event::BlockEnd(ampd_proto::EventBlockEnd { height });
+
+        let domain_event_response = Event::try_from(proto_event);
+        assert!(domain_event_response.is_ok());
+        let domain_event = domain_event_response.unwrap();
+
+        assert!(
+            matches!(domain_event, Event::BlockEnd(h) if h == block::Height::try_from(height).unwrap())
+        );
+    }
+
+    #[test]
+    fn abci_event_conversion_should_succeed() {
+        let contract_address_string =
+            "axelarvaloper1zh9wrak6ke4n6fclj5e8yk397czv430ygs5jz7".to_string();
+
+        let mut attrs = HashMap::new();
+        attrs.insert("key1".to_string(), "value1".to_string());
+        attrs.insert("key2".to_string(), "42".to_string());
+        attrs.insert(
+            "_contract_address".to_string(),
+            contract_address_string.clone(),
+        );
+
+        let proto_event = ampd_proto::subscribe_response::Event::Abci(ampd_proto::Event {
+            r#type: "test_event".to_string(),
+            contract: contract_address_string.clone(),
+            attributes: attrs,
+        });
+
+        let domain_event_response = Event::try_from(proto_event.clone());
+        assert!(domain_event_response.is_ok());
+        let domain_event = domain_event_response.unwrap();
+
+        goldie::assert!(&domain_event.to_string());
+
+        let converted_proto_res = ampd_proto::Event::try_from(domain_event);
+        assert!(converted_proto_res.is_ok());
+        let converted_proto = converted_proto_res.unwrap();
+
+        assert_eq!(converted_proto.r#type, "test_event");
+        assert_eq!(converted_proto.contract, contract_address_string,);
+        assert_eq!(
+            converted_proto.attributes.get("key1").unwrap(),
+            "\"value1\""
+        );
+        assert_eq!(converted_proto.attributes.get("key2").unwrap(), "42");
```

Comment:
I'd rather split this into a second test, so it can be a golden file test as well